core/primitives-core/src/account.rs

```diff
@@ -67,13 +68,49 @@ impl AccountV1 {
         AccountV2 {
             amount: self.amount,
             locked: self.locked,
-            code_hash: self.code_hash,
             storage_usage: self.storage_usage,
+            account_contract: convert_code_hash(self.code_hash),
         }
     }
 }
 
-// TODO(global-contract): add new field
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    PartialEq,
+    Eq,
+    Debug,
+    Clone,
+    ProtocolSchema,
+)]
+pub enum AccountContractType {
+    None,
+    LocalContract(CryptoHash),
+    GlobalContract(CryptoHash),
+    GlobalContractByAccount(AccountId, CryptoHash),
+}
+
+impl From<AccountContractType> for CryptoHash {
```

Comment:
`From` trait is usually implemented when it is applicable for the whole type. In our case there is not generic conversion from `AccountContractType` to any `CryptoHash`, only to the hash representing local contract code, so in this case it is better to just implement member function `fn to_local_contract_code(&self) -> CryptoHash`